gradio/components/deep_link_button.py

```diff
@@ -0,0 +1,129 @@
+"""Predefined button to copy a shareable link to the current Gradio Space."""
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from pathlib import Path
+from typing import TYPE_CHECKING, Literal
+
+from gradio_client.documentation import document
+
+from gradio import utils
+from gradio.components import Button, Component
+from gradio.context import get_blocks_context
+
+if TYPE_CHECKING:
+    from gradio.components import Timer
+
+
+@document()
+class DeepLinkButton(Button):
+    """
+    Creates a button that copies a shareable link to the current Gradio Space.
+    The link includes the current session hash as a query parameter.
+    """
+
+    is_template = True
+    n_created = 0
+
+    def __init__(
+        self,
+        value: str = "Share via Link",
+        copied_value: str = "Link Copied!",
+        *,
+        inputs: Component | Sequence[Component] | set[Component] | None = None,
```

Comment:
What's the purpose of `inputs`?

Reply:
Just following the convention for all components. They should take inputs in case value is a callable but it doesn't really make sense in this case.